portfolio/src/main/webapp/index.html

```diff
@@ -8,39 +8,72 @@
   </head>
   <body>
     <div id="content">
+      <div id="menu">
+        <h2>
+        <a href="index.html">Home</a>&nbsp&nbsp
+        <a href="https://ca.linkedin.com/in/jennifer-dai">LinkedIn</a>&nbsp&nbsp
+        <a href="https://github.com/nif33">Github</a>&nbsp&nbsp
+        </h2>
+      </div>
+
       <div id="header">
         <h1>Jennifer Dai's Portfolio</h1>
-        <a href="https://ca.linkedin.com/in/jennifer-dai">LinkedIn</a>
-        <a href="https://github.com/nif33">Github</a>
       </div>
 
       <p>Click here to get a random fact about me:</p>
-      <button onclick="addRandomFact()">Tell me more!</button>
-      <div id="fact-container"></div>
+        <button onclick="addRandomFact()">Tell me more!</button>
+        <marquee><div id="fact-container"></div></marquee>
 
       <h2>Background</h2>
-      <p>I'm a 2nd year Computing Science student at Simon Fraser University. Three fun facts about me: 
-          I have a pet dog named William, my favourite place I've travelled to is Montreal, 
-          and my first day of work at Google was also my 20th birthday!
+      <p>
+        I'm a 2nd year Computing Science student at Simon Fraser University. Three fun facts about me: 
+        I have a pet dog named William, my favourite place I've travelled to is Montreal, 
+        and my first day of work at Google was also my 20th birthday!
       </p>
 
       <h2>Photos of my Dog</h2>
-      <p>This is my dog: William</p>
-      <div>
+      <p>
+        This is my dog: William. People always tell me his name is too human and I know, but I did not choose his name. 
+        He is a 7 year old Miniaature Poodle. His hobbies are taking naps, eating snacks, and barking at firetrucks.
+      </p>
+      <div class="slideshow">
         <img class="dog-slides" src="/images/dog/dog0.jpg">
         <img class="dog-slides" src="/images/dog/dog1.jpg">
         <img class="dog-slides" src="/images/dog/dog2.jpg">
+        
+        <button onclick="nextSlide(0, -1)">Previous</button>
+        <button onclick="nextSlide(0, 1)">Next</button>
```

Comment:
Having JS looking calls in HTML seems a little bit messy. The thing that bothers me the most is explicit arguments for the function in HTML. Not too familiar with this but maybe have a function called nextDogSlide(), prevDogSlide(), nextPlaceSlide(), and prevPlaceSlide() and in their definitions call another function that actually does the work with the specific params. Would like to see what Andrew thinks about this.

On the same topic of JS functions calls in HTML, maybe you should be doing this fully in JS. This is what I found online
https://stackoverflow.com/questions/5871640/why-is-using-onclick-in-html-a-bad-practice
https://www.reddit.com/r/javascript/comments/3kamqn/why_is_inline_onclickfunction_considered_such_a/

So a solution can be something like what is in the stackoverflow link or something like

HTML:
`<button id="next-dog-slide-button">Previous</button>`

JS:
```
document.getElementById('next-dog-slide-button').onclick = function() {
  // Write how to handle this click here
}
```

Again, I would like Andrew's take on this.

Reply:
One term you'll hear sometimes is "[code smell](https://en.wikipedia.org/wiki/Code_smell)." I would say Juan is picking up on a code smell here.

So why is it "smelly" to have these function calls with explicit arguments in HTML? I can think of a few reasons:

- It's not obvious at a glance what the different arguments mean.

- It's hard to catch typos; since these function calls are just strings in HTML your editor won't highlight syntax errors

- Most important, in my opinion: it's hard to modify. Imagine you want to add a new slide show between dog photos and place photos. After you add your new slideshow div you need to remember to update the function arguments for the place photos and food photos (because now they are the 3rd and 4th slideshows instead of 2nd and 3rd.)

Overall: this solution works, but I think we can design a solution that (1) works just as well and (2) is easier to reason about and modify. I think Juan's suggestions about moving more logic to JS are a good starting point.